test/e2eprovider/server/server.go

```diff
@@ -0,0 +1,144 @@
+package e2eprovider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+
+	util "sigs.k8s.io/secrets-store-csi-driver/pkg/csi-common"
+	"sigs.k8s.io/secrets-store-csi-driver/provider/v1alpha1"
+	types "sigs.k8s.io/secrets-store-csi-driver/test/e2eprovider/types"
+	vault "sigs.k8s.io/secrets-store-csi-driver/test/e2eprovider/vault"
+
+	"google.golang.org/grpc"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/yaml"
+)
+
+// Server is a mock csi-provider server
+type Server struct {
+	grpcServer *grpc.Server
+	listener   net.Listener
+	SocketPath string
+	network    string
+	Vault      vault.Vault
+}
+
+// NewE2eProviderServer returns a mock csi-provider grpc server
```

Comment:
Let's standardize using mock/fake across all the changes here

Reply:
updated to `mock`